### DISCUSSION TITLE
Not possible to have an upload to be optional, is it a bug?

### DISCUSSION THREAD

**Discussion by hunghvu** [payloadcms/payload#1023] [Q&A] [answered]
When I test the upload feature, it seems like an upload field is not optional. For example,

``` ts
import { CollectionConfig } from "payload/types";

const Test:CollectionConfig = {
  slug: 'tests',
  fields: [
      {
        name: 'metaCoverImage',
        label: 'Cover Image',
        type: 'upload',
        relationTo: 'media',
        required: true,
      },
      {
        name: 'metaCustomOgImage',
        label: 'Custom OG Image (front-end will fallback to cover image if not presented)',
        type: 'upload',
        relationTo: 'media',
      },
    ]
}

export default Test;
```

The second upload field should be optional, however, it raises an exception when no upload is presented. Is this a design decision or a bug? If it is a design decision, then is there a way to have an optional upload?

![image](https://user-images.githubusercontent.com/62678592/187333956-655bb28e-116e-4a74-b0c7-4e77a01f035d.png)

**Comment by DanRibbens** [comment, marked as answer]
Hey @hunghvu, it looks like a regression from the changes we made to futher validate upload and relationships. I'll get a fix out soon!

**Reply by DanRibbens** [reply]
OK, this is fixed now, please upgrade to v1.0.30.

Thanks for asking or I wouldn't have caught it!

**Reply by hunghvu** [reply]
Can confirm that it was fixed. I always love this lightning-fast response from the team. Keep up the great work!
